client/src/test/java/io/pravega/client/stream/impl/SegmentSelectorTest.java

```diff
@@ -109,4 +121,46 @@ public void testSameRoutingKey() {
         assertArrayEquals(new int[] { 20, 0, 0, 0 }, counts);
     }
 
+    @Test
+    public void testStreamDeletion() {
+        final Segment segment0 = new Segment(scope, streamName, 0);
+        final Segment segment1 = new Segment(scope, streamName, 1);
+        final CompletableFuture<Void> writerFuture = new CompletableFuture<>();
+
+        // Setup Mock.
+        SegmentOutputStream s0Writer = Mockito.mock(SegmentOutputStream.class);
+        SegmentOutputStream s1Writer = Mockito.mock(SegmentOutputStream.class);
+        when(s0Writer.getUnackedEventsOnSeal())
+                .thenReturn(ImmutableList.of(new PendingEvent("0", ByteBuffer.wrap("e".getBytes()), writerFuture)));
+
+        SegmentOutputStreamFactory factory = Mockito.mock(SegmentOutputStreamFactory.class);
+        when(factory.createOutputStreamForSegment(eq(segment0), ArgumentMatchers.<Consumer<Segment>>any(), any(EventWriterConfig.class), anyString()))
+                .thenReturn(s0Writer);
+        when(factory.createOutputStreamForSegment(eq(segment1), ArgumentMatchers.<Consumer<Segment>>any(), any(EventWriterConfig.class), anyString()))
+                .thenReturn(s1Writer);
+
+        Controller controller = Mockito.mock(Controller.class);
+        SegmentSelector selector = new SegmentSelector(new StreamImpl(scope, streamName), controller, factory, config);
+        TreeMap<Double, Segment> segments = new TreeMap<>();
+        segments.put(0.5, segment0);
+        segments.put(1.0, segment1);
+        StreamSegments streamSegments = new StreamSegments(segments, "");
+
+        when(controller.getCurrentSegments(scope, streamName))
+                .thenReturn(CompletableFuture.completedFuture(streamSegments));
+        //trigger refresh.
+        selector.refreshSegmentEventWriters(segmentSealedCallback);
+
+        //simulate stream deletion where controller.getSuccessors() is completed exceptionally.
+        when(controller.getSuccessors(segment0))
+                .thenAnswer(i -> {
+                    CompletableFuture<StreamSegmentsWithPredecessors> result = new CompletableFuture<>();
+                    result.completeExceptionally(new RuntimeException());
```

Comment:
Can we use the precise exception that is supposed to be thrown, primarily for education purposes?

Reply:
The controller throws `io.pravega.controller.store.stream.StoreException$DataNotFoundException` which is type RuntimeException. Using RuntimeException as it is not visible to the client.
I have documented the exception in the test.